### PR TITLE
Add quad VTK_LAGRANGE elements

### PR DESCRIPTION
**Pull request by alexfikl**
Adds the quadrilateral (2D) and hexahedron (3D) elements.

### CODE REVIEW

**Comment by alexfikl** [comment]
This probably needs a big "Works For Me" sticker on it. 

I've been looking through VTK/Paraview patches that change the ordering of some of these things and they seem to be backporting some of them, so I have no idea if it works on anything else besides Paraview 5.8.1.

File: pyvisfile/vtk/vtk_ordering.py [hunk 1]
```diff
@@ -210,7 +209,7 @@ def vtk_lagrange_simplex_node_tuples(dims, order, is_consistent=False):
 
 
 def vtk_lagrange_simplex_node_tuples_to_permutation(node_tuples):
-    order = max([sum(i) for i in node_tuples])
+    order = max([max(i) for i in node_tuples])
```
Comment (inducer) [inline pyvisfile/vtk/vtk_ordering.py:212]: I agree that these two are equivalent, but I'm still curious why this change happened, especially given that the function name mentions "simplex".

**Review by inducer — APPROVED** [review]
Thanks for working on this! LGTM, but I've got one minor question.

File: pyvisfile/vtk/vtk_ordering.py (hunk 1 above)
Comment (alexfikl) [reply pyvisfile/vtk/vtk_ordering.py:212]: Mostly just to have it consistent with the quad one below. I initially copy pasted it and it obviously didn't work there.

File: pyvisfile/vtk/vtk_ordering.py [hunk 2]
```diff
@@ -59,13 +64,8 @@ def add_tuple_to_list(ary, x):
...
 
-def vtk_lagrange_curve_node_tuples(order, is_consistent=False):
-    if is_consistent:
-        node_tuples = [(0,), (order,)] + [(i,) for i in range(1, order)]
-    else:
-        node_tuples = [(i,) for i in range(order + 1)]
-
-    return node_tuples
+def vtk_lagrange_curve_node_tuples(order):
+    return [(0,), (order,)] + [(i,) for i in range(1, order)]
```
Comment (alexfikl) [inline pyvisfile/vtk/vtk_ordering.py:68]: To add some more explanations here. The `is_consistent` stuff was removed because it turned out to be a bug in the `vtkCellTypeSource` function I was using to check if we have the same order and not in the reader/writer/element definition.

File: pyvisfile/vtk/vtk_ordering.py [hunk 3]
```diff
@@ -182,25 +182,24 @@ def vtk_lagrange_tetrahedron_node_tuples(order):
...
     """
     :arg dims: dimension of the simplex, i.e. 1 corresponds to a curve, 2 to
         a triangle, etc.
     :arg order: order of the polynomial representation, which also defines
         the number of nodes on the simplex.
-    :arg is_consistent: If *True*, 1D curve node ordering will follow the
-        same rules as the higher-dimensional simplices by putting the
-        vertices first. This is not the default in VTK as of version 8.1,
-        when the higher-order elements were introduced.
+    :arg vtk_version: a :class:`tuple` of two elements containing the version
```
Comment (alexfikl) [inline pyvisfile/vtk/vtk_ordering.py:192]: I left the `vtk_version` here, even though it's no necessary anymore for simplices, for consistency with the quad routine and in case they change things in the future.

**Comment by inducer** [comment]
LGTM. Thanks for working on this!
